### PR TITLE
Revising the toString methods for descendants of M3Node

### PR DESCRIPTION
**Pull request by ftomassetti**
They now all produce the same initial part and some add additional details

### CODE REVIEW

File: core/src/main/java/io/lionweb/lioncore/java/language/Containment.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public void setSpecialized(@Nullable Containment specialized) {
 
   @Override
   public String toString() {
-    return "Containment{" + "name=" + getName() + ", " + "type=" + getType() + '}';
+    return super.toString() + "{" + "name=" + getName() + ", " + "type=" + getType() + '}';
```
Comment (enikao) [inline core/src/main/java/io/lionweb/lioncore/java/language/Containment.java:103]: This is the same implementation as for `Reference`, so we can move it to `Link`

File: core/src/main/java/io/lionweb/lioncore/java/language/PrimitiveType.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public PrimitiveType(@Nullable Language language, @Nullable String name, String
 
   @Override
   public String toString() {
-    return "PrimitiveType(" + getName() + ")";
+    return super.toString() + "{" + "name=" + getName() + '}';
```
Comment (enikao) [inline core/src/main/java/io/lionweb/lioncore/java/language/PrimitiveType.java:40]: This is the only `LanguageEntity` that has a special `toString()`. Why?

File: core/src/main/java/io/lionweb/lioncore/java/language/Containment.java (hunk 1 above)
Comment (enikao) [reply core/src/main/java/io/lionweb/lioncore/java/language/Containment.java:103]: Why don't construct a _qualifiedName_ like for `LanguageEntity` (for symmetry)?

File: core/src/main/java/io/lionweb/lioncore/java/language/Property.java [hunk 3]
```diff
@@ -73,7 +73,7 @@ public Property setType(@Nullable DataType type) {
 
   @Override
   public String toString() {
-    return "Property{" + "name=" + getName() + ", " + "type=" + getType() + '}';
+    return super.toString() + "{" + "name=" + getName() + ", " + "type=" + getType() + '}';
```
Comment (enikao) [inline core/src/main/java/io/lionweb/lioncore/java/language/Property.java:76]: Why don't construct a _qualifiedName_ like for `LanguageEntity` (for symmetry)?

File: core/src/main/java/io/lionweb/lioncore/java/language/PrimitiveType.java (hunk 2 above)
Comment (ftomassetti) [reply core/src/main/java/io/lionweb/lioncore/java/language/PrimitiveType.java:40]: Right, let's remove it

File: core/src/main/java/io/lionweb/lioncore/java/language/Containment.java (hunk 1 above)
Comment (ftomassetti) [reply core/src/main/java/io/lionweb/lioncore/java/language/Containment.java:103]: Moving to Link, and using qualifiedName with the same logic used in LanguageEntity. I will also use qualifiedName in Property.toString

File: core/src/main/java/io/lionweb/lioncore/java/language/Property.java (hunk 3 above)
Comment (ftomassetti) [reply core/src/main/java/io/lionweb/lioncore/java/language/Property.java:76]: Done

**Comment by ftomassetti** [comment]
Thank you @enikao ! I applied all of your suggestions and I will now merge this

**Pull request merged**